validator/accounts/accounts_list.go

```diff
@@ -4,50 +4,27 @@ import (
 	"context"
 	"fmt"
 	"math"
-	"strings"
 
 	"github.com/pkg/errors"
-	"github.com/prysmaticlabs/prysm/cmd/validator/flags"
 	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
-	"github.com/prysmaticlabs/prysm/validator/accounts/iface"
 	"github.com/prysmaticlabs/prysm/validator/accounts/wallet"
 	"github.com/prysmaticlabs/prysm/validator/keymanager"
-	"github.com/urfave/cli/v2"
 )
 
-// ListAccountsCli displays all available validator accounts in a Prysm wallet.
-func ListAccountsCli(cliCtx *cli.Context) error {
-	w, err := wallet.OpenWalletOrElseCli(cliCtx, func(cliCtx *cli.Context) (*wallet.Wallet, error) {
-		return nil, wallet.ErrNoWalletFound
-	})
-	if err != nil {
-		return errors.Wrap(err, "could not open wallet")
-	}
-	// TODO(#9883) - Remove this when we have a better way to handle this. this is fine.
-	// genesis root is not set here which is used for sign function, but fetch keys should be fine.
-	km, err := w.InitializeKeymanager(cliCtx.Context, iface.InitKeymanagerConfig{ListenForChanges: false})
-	if err != nil && strings.Contains(err.Error(), keymanager.IncorrectPasswordErrMsg) {
-		return errors.New("wrong wallet password entered")
-	}
-	if err != nil {
-		return errors.Wrap(err, ErrCouldNotInitializeKeymanager)
-	}
-	showDepositData := cliCtx.Bool(flags.ShowDepositDataFlag.Name)
-	showPrivateKeys := cliCtx.Bool(flags.ShowPrivateKeysFlag.Name)
-	listIndices := cliCtx.Bool(flags.ListValidatorIndices.Name)
-
-	if listIndices {
-		client, _, err := prepareClients(cliCtx)
+// List pretty-prints accounts in the wallet.
+func (acm *AccountsCLIManager) List(ctx context.Context) error {
+	if acm.listValidatorIndices {
+		client, _, err := acm.prepareBeaconClients(ctx)
 		if err != nil {
 			return err
 		}
-		return listValidatorIndices(cliCtx.Context, km, *client)
+		return listValidatorIndices(ctx, acm.keymanager, *client)
 	}
-	return km.ListKeymanagerAccounts(cliCtx.Context,
+	return acm.keymanager.ListKeymanagerAccounts(ctx,
```

Comment:
i wonder if we needed to do a type check if the keymanager supports this , probably an issue with the old code too 🤔  > ListKeymanagerAccounts. I don't think all keymanager types support this.

Reply:
I don't think we need to do that type assertion stuff anymore actually. In validator/keymanager/types.go we enforce that `ListKeymanagerAccounts` is part of the IKeymanger interface. That was the point of https://github.com/prysmaticlabs/prysm/pull/10382. Basically if the keymanager type doesn't support the List, then it will just always return an error and let the caller deal with it.